### PR TITLE
[AIR] Deprecate `ray.tune.logger.Logger` interface

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

This PR deprecates:
1. Soft-deprecated, for removal in 2.7.
    - `ray.tune.logger.Logger`, in favor of `ray.tune.logger.LoggerCallback`
         - Also, deprecated any built-in `Logger` subclasses, including `CSVLogger`, `JsonLogger`, `TBXLogger`, etc.

### See also

- See the docs under "Deprecations" in [this doc](https://docs.google.com/document/d/1s2VouZg1CtzOCviZMHFFDPVMdJAPeXrTNcsBftwXmkQ/edit#) for context.
- Also see [this tracker spreadsheet](https://docs.google.com/spreadsheets/d/1rhMOGB1EwpEuLUu0BlBdzj28V2Wav96VqgnIIHJtLF8/edit).

If you don't have access to these docs, please feel free to leave a comment on this PR, and I can provide the context for these deprecations!

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/integration/comet.py [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-@Deprecated(message=callback_deprecation_message)
-class CometLoggerCallback(_CometLoggerCallback):
-    def __init__(
-        self,
-        online: bool = True,
-        tags: List[str] = None,
-        save_checkpoints: bool = False,
-        **experiment_kwargs
-    ):
-        logging.warning(callback_deprecation_message)
```
Comment (krfricke) [inline python/ray/tune/integration/comet.py:27]: Instead of removing, should we raise an exception here?

Comment (justinvyu) [reply python/ray/tune/integration/comet.py:27]: Ah yeah, technically it's been deprecated for like 4 releases, so I was going to skip directly to removal, but we can just hard deprecate then remove in 2.6.

So, the schedule is 2 releases of soft deprecation, then 1 release of hard deprecation, then removal?

Comment (krfricke) [reply python/ray/tune/integration/comet.py:27]: For public APIs, yes! For beta and DeveloperAPIs, 1 release of soft deprecation is sufficient.
I think we should just always do the error raising - people may ignore warnings (especially if scripts are run in jobs/on schedules) and the raise will make sure they can migrate their scripts easily.

**Review by krfricke — APPROVED** [review]
Great!

**Pull request merged**